src/actions/proxy.ts

```diff
@@ -102,16 +99,15 @@ export function ProxyActions<ActionsBase extends Constructor<TinlakeParams>>(Bas
             { type: 'uint256', name: 'token' },
           ],
         },
-        [this.contracts['SHELF'].address, nftRegistryAddr, tokenId]
+        [this.contract('SHELF').address, nftRegistryAddress, tokenId]
       )
 
-      const txHash = await executeAndRetry(proxy.execute, [this.contracts['ACTIONS'].address, encoded, this.ethConfig])
-      console.log(`[Proxy Transfer Issue Loan] txHash: ${txHash}`)
-      return waitAndReturnEvents(this.eth, txHash, this.contractAbis['PROXY'], this.transactionTimeout)
+      return this.pending(proxy.execute(this.contract('ACTIONS').address, encoded, this.overrides))
     }
 
-    proxyLockBorrowWithdraw = async (proxyAddr: string, loanId: string, amount: string, usr: string) => {
-      const proxy: any = this.eth.contract(this.contractAbis['PROXY']).at(proxyAddr)
+    proxyLockBorrowWithdraw = async (proxyAddress: string, loanId: string, amount: string, usr: string) => {
+      const proxy = this.contract('PROXY', proxyAddress)
+      // TODO: replace with ethers.js method call
```

Comment:
Still open?

Reply:
Yes, but the web3 `abiCoder` works just fine, so I'd say it's fine to address that later, if you're OK with that.

Reply:
Sure, can leave it as is then. Thanks!